salt/modules/win_pkg.py

```diff
@@ -426,80 +426,252 @@ def _get_reg_software():
-                   'SchedulingAgent',
-                   'WIC',
-                   'Not Found',
-                   '(value not set)',
-                   '',
-                   None]
-
+    # Logic for this can be found in this question:
+    # https://social.technet.microsoft.com/Forums/windows/en-US/d913471a-d7fb-448d-869b-da9025dcc943/where-does-addremove-programs-get-its-information-from-in-the-registry
     reg_software = {}
-    hive = 'HKLM'
-    key = "Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall"
-
-    def update(hive, key, reg_key, use_32bit):
-        # 2018 this code has been update to reflect some of utils/pkg/win.py logic
-        # i.e. check_reg, and checking of DisplayName and DisplayVersion.
-        d_name = ''
-        d_vers = ''
-
-        d_name_regdata = __salt__['reg.read_value'](hive,
-                                                    '{0}\\{1}'.format(key, reg_key),
-                                                    'DisplayName',
-                                                    use_32bit)
+
+    def skip_component(hive, key, reg_key, use_32bit):
+        '''
+        Returns:
+            bool: True if the component needs to be skipped, otherwise False
+        '''
+        # Skip SystemComponents
+        # Skip WindowsInstaller (removes Compiler for Python 2.7)
+        #for check in ['SystemComponent', 'WindowsInstaller']:
+        for check in ['SystemComponent']:
+            if __utils__['reg.value_exists'](
+                    hive=hive,
+                    key='{0}\\{1}'.format(key, reg_key),
+                    vname=check,
+                    use_32bit_registry=use_32bit):
+                if __utils__['reg.read_value'](
+                        hive=hive,
+                        key='{0}\\{1}'.format(key, reg_key),
+                        vname=check,
+                        use_32bit_registry=use_32bit)['vdata'] > 0:
+                    return True
+        return False
+
+    def skip_additional_checks(hive, key, reg_key, use_32bit):
+        '''
+        Returns:
+            bool: True if the component needs to be skipped, otherwise False
+        '''
+        # Must contain one of UninstallString, QuietUninstallString, ModifyPath
+        # If it does not exist, skip (return True)
+        vname_present = False
+        for reg_vname in ['UninstallString',
+                          'QuietUninstallString',
+                          'ModifyPath']:
+            if __utils__['reg.value_exists'](
+                    hive=hive,
+                    key='{0}\\{1}'.format(key, reg_key),
+                    vname=reg_vname,
+                    use_32bit_registry=use_32bit):
+                vname_present = True
+                break
+        if not vname_present:
+            return True
+
+        # Check for Hotfixes, Service Packs, etc...
+        # If found, skip (return True)
+        skip_types = ['Hotfix',
+                      'Service Pack',
```

Comment:
If you look at salt/utils/pkg/win.py it has  ('Hotfix', 'Update Rollup', 'Security Update', 'ServicePack')
Keep in mind this change is a behavior change hence I have never excluded it, just encase someone is using it.

Reply:
I made it optional. Added some parameters that default to existing behavior.